### DISCUSSION TITLE
SQLModel Relationships not present in Pydantic model_dump

### DISCUSSION THREAD

**Discussion by konkyrkos** [fastapi/sqlmodel#1092] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Club(SQLModel, table=True):  
    __tablename__ = "clubs"

    updated_at: datetime = Field(sa_type=DateTime, sa_column_kwargs={"server_default": utcnow()})
    id: Optional[int] = Field(nullable=True, index=True, unique=True)
    uid: str = Field(nullable=False, sa_type=String, primary_key=True, sa_column_kwargs={"autoincrement":False})
    name: Optional[str] = Field(default=None, nullable=True)

    members: List[Member] = Relationship(back_populates="club", sa_relationship_kwargs=dict(cascade="all"))

class Member(SQLModel, table=True):  # type: ignore
    __tablename__ = "members"
    id: int = Field(nullable=False, sa_type=Integer, primary_key=True, sa_column_kwargs={
        "autoincrement": False
    })
    club_uid: str = Field(nullable=False, foreign_key="clubs.uid", index=True)

    club: "Club" = Relationship(back_populates="members")
```

### Description

We are trying to update to pydantic v2 (from v1) and latest versions of sqlmodel and sqlalchemy. We have two models connected by a relationship. We want to get a model as a dict along with its relationships (as keys of the dict too). We use club.model_dump(inlcude={'id', 'uid', 'name', 'members'}). In the past instead of .model_dump we used .dict() with the same parameters it worked. Now model_dump method does not work and does not return the relationships as keys of the dict. We have tried various options of the method but nothing did the trick. 
Any help and interest in the issue is welcome!
Thank you in advance.

### Operating System

macOS

### SQLModel Version

0.0.21

### Python Version

3.10.5

**Comment by theblazehen** [comment]
@konkyrkos Any chance you may have figured this out?

**Comment by monchin** [comment]
Would you please offer a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? An example that uses sqlite and can be directly run, and print the result, and you can tell us what result you expected.

I guess this is because sqlmodel dosen't validate data, so maybe #1041 could work, but I need an example to try.

**Comment by madduck** [comment]
Here is a minimal example:

```Python
from sqlmodel import Field, Relationship, SQLModel


class ParentBase(SQLModel):
    name: str | None


class Parent(ParentBase, table=True):
    id: int | None = Field(default=None, primary_key=True)
    children: list["Child"] = Relationship(back_populates="parent")


class ChildBase(SQLModel):
    name: str
    parentid: int | None = Field(default=None, foreign_key="parent.id")


class Child(ChildBase, table=True):
    id: int | None = Field(default=None, primary_key=True)
    parent: Parent = Relationship(back_populates="children")


class ChildWithParent(ChildBase):
    id: int
    parent: Parent


if __name__ == "__main__":
    p1 = Parent(id=1, name="parent")

    c1 = Child(id=1, parent=p1, name="child1")
    dump = c1.model_dump()
    print(dump)

    c2 = ChildWithParent(id=2, parent=p1, name="child2")
    dump = c2.model_dump()
    print(dump)
```

The only difference between instances of `ChildWithParents` and `Child` models are that in the latter, the `parent` attribute is default-initialised with a `sqlmodel.Relationship`. This seems to effect that the Pydantic serialiser doesn't care about the field anymore, and it doesn't include it in the model dump.

I encountered this issue over at [FastAPI](https://github.com/fastapi/fastapi/discussions/13856), and I am glad to be able to further hone down on the problem.

Unless there is a necessary reason for why a field that defaults to a `Relationship` cannot be part of the Pydantic model, I think it would be great to get that fixed. Because it is one of the core promises of `SQLModel` that each model is also a Pydantic model, but in this case, I am not sure it is.

**Comment by ttlns** [comment]
Here's a workaround for serializing (https://github.com/fastapi/sqlmodel/discussions/1092#discussioncomment-13665469 for reference):

```python

from typing import Callable, Any

class Child(ChildBase, table=True):
    id: int | None = Field(default=None, primary_key=True)
    parent: Parent = Relationship(back_populates="children")

    @model_serializer(mode="wrap")
    def serialize_model(self, handler:Callable[[Any], dict]):
        data = handler(self)
        if self.parent is not None:
            data["parent"]=self.parent.model_dump()
        return data
```
This could as well be written recustively foreach object and check for `Relationship` attributes.

This will of course not work for `Child.model_validate(resp_dict)`. Unfortunately, even custom [model-validators](https://docs.pydantic.dev/latest/concepts/validators/#model-validators) in `wrap` mode don't have access to the dict and resulting model inside a single callback.
